### DISCUSSION TITLE
Potential bug in 3.2.0 - x-data exposed as 'data'

### DISCUSSION THREAD

**Discussion by blackmagic0** [alpinejs/alpine#1664] [7. Bugs (archived)]
We noticed a potential bug appear between 523a65234031340b89fafe2752b727544a6fa801 and 4daafb443c3dfd50b5a91879897ea1c6c3aad0e4 where data is being exposed to components as the entire x-data object for the component.  

See codepen:
-> in 3.1.1 data does not have content: https://codepen.io/jacobpozaic/pen/PomomwZ
-> in 3.2.0 data is accessible and matches x-data: https://codepen.io/jacobpozaic/pen/XWRWMvq

was this an intended change?

**Comment by danddanddand** [comment]
Our app fails on 3.2.0 as well:
```
alpinejs.v3.2.0.js:2772 Uncaught ReferenceError: data is not defined
    at alpinejs.v3.2.0.js:2772
    at Function.<anonymous> (alpinejs.v3.2.0.js:1863)
    at flushHandlers (alpinejs.v3.2.0.js:1633)
    at stopDeferring (alpinejs.v3.2.0.js:1637)
    at deferHandlingDirectives (alpinejs.v3.2.0.js:1640)
    at initTree (alpinejs.v3.2.0.js:1824)
    at alpinejs.v3.2.0.js:1795
    at Array.forEach (<anonymous>)
    at Object.start (alpinejs.v3.2.0.js:1794)
    at init-snowpack.js:292
(anonymous)	@	alpinejs.v3.2.0.js:2772
(anonymous)	@	alpinejs.v3.2.0.js:1863
flushHandlers	@	alpinejs.v3.2.0.js:1633
stopDeferring	@	alpinejs.v3.2.0.js:1637
deferHandlingDirectives	@	alpinejs.v3.2.0.js:1640
initTree	@	alpinejs.v3.2.0.js:1824
(anonymous)	@	alpinejs.v3.2.0.js:1795
start	@	alpinejs.v3.2.0.js:1794
(anonymous)	@	init-snowpack.js:292
```

**Comment by ryangjchandler** [comment]
@JacobPozaic I've pinged @calebporzio so hopefully he sees this soon, quick on the PR too, nice work!

**Reply by danddanddand** [reply]
It's funny how we noticed it. Resolving merge conflict with his changes this morning and when we ran our tests the x-model one line 99 here started failing because data name we used randomly was intended as reference error... We were like, oh no, what did we do... 

https://github.com/alpinejs/alpine/pull/1607/files#diff-fae1e2641a98edadbc8ec90b1851152ce1725dc2fae9b89eb2363c5163b0573c

```
test('x-model identifier issue',
    [html`
        <div x-data="{value: ''}">
            <div id="errors">false</div>
            <input id="xmodel" x-model="data"/>
        </div>
    `,
        setupConsoleInterceptor( "xmodel" )
    ],
    ({ get }) => {
        get( "#errors" ).should(haveText('true'))
    }
)
```
